test/e2e/apps/job.go

```diff
@@ -344,6 +344,53 @@ var _ = SIGDescribe("Job", func() {
 		}
 	})
 
+	ginkgo.It("should recreate pods only after they have failed if pod replacement policy is set to Failed", func(ctx context.Context) {
+		ginkgo.By("Creating a job")
+		job := e2ejob.NewTestJob("notTerminate", "pod-recreate-failed", v1.RestartPolicyNever, 1, 1, nil, backoffLimit)
+		job.Spec.PodReplacementPolicy = ptr.To(batchv1.Failed)
+		job.Spec.Template.Spec.Containers[0].Image = "gcr.io/k8s-staging-perf-tests/sleep"
+		job.Spec.Template.Spec.Containers[0].Args = []string{"30s", "-termination-grace-period", "1m"}
+		job, err := e2ejob.CreateJob(ctx, f.ClientSet, f.Namespace.Name, job)
+		framework.ExpectNoError(err, "failed to create job in namespace: %s", f.Namespace.Name)
+
+		ginkgo.By("Deleting job pod")
+		err = e2ejob.WaitForJobPodsRunning(ctx, f.ClientSet, f.Namespace.Name, job.Name, parallelism)
+		framework.ExpectNoError(err, "failed to ensure number of pods associated with job %s is equal to parallelism count in namespace: %s", job.Name, f.Namespace.Name)
+
+		pods, err := e2ejob.GetJobPods(ctx, f.ClientSet, f.Namespace.Name, job.Name)
+		framework.ExpectNoError(err, "failed to get pod list for job %s in namespace: %s", job.Name, f.Namespace.Name)
+
+		framework.ExpectNoError(e2epod.DeletePodsWithGracePeriod(ctx, f.ClientSet, pods.Items, 10), "failed to delete pods in namespace: %s", f.Namespace.Name)
+
+		framework.ExpectNoError(e2epod.WaitForPodTerminatingInNamespaceTimeout(
+			ctx, f.ClientSet, pods.Items[0].Name, f.Namespace.Name, f.Timeouts.PodDelete),
+			"failed to wait for pod to be terminating in namespace: %s", f.Namespace.Name,
+		)
```

Comment:
We don't need to check this. It's beyond the responsibility of the job controller and I'm sure it's covered by tests focused on Pod.